src/plugins/discover/public/application/main/components/top_nav/open_search_panel.test.tsx

```diff
@@ -8,25 +8,42 @@
 
 import React from 'react';
 import { shallow } from 'enzyme';
+import { OpenSearchPanel } from './open_search_panel';
 
-const mockCapabilities = jest.fn().mockReturnValue({
-  savedObjectsManagement: {
-    edit: true,
-  },
-});
-
-jest.mock('../../../../kibana_services', () => {
+jest.mock('../../../../utils/use_discover_services', () => {
+  const defaults = {
+    core: { uiSettings: {}, savedObjects: {} },
+    addBasePath: (path: string) => path,
+  };
+  let testNumber = 0;
   return {
-    getServices: () => ({
-      core: { uiSettings: {}, savedObjects: {} },
-      addBasePath: (path: string) => path,
-      capabilities: mockCapabilities(),
-    }),
+    useDiscoverServices: () => {
+      testNumber++;
+
+      if (testNumber === 2) {
+        return {
+          ...defaults,
+          capabilities: {
+            savedObjectsManagement: {
+              edit: false,
+              delete: false,
+            },
+          },
```

Comment:
Implementing it this way means, the tests can only be run in the given specific number, the 2nd test can't be run separately. We should omit this pattern, and mock the return of `savedObjectsManagement` separately in each test

Reply:
Changed.